camel-core/src/test/java/org/apache/camel/processor/ThrottlingExceptionRoutePolicyKeepOpenOnInitTest.java

```diff
@@ -0,0 +1,98 @@
+package org.apache.camel.processor;
+
+import org.apache.camel.ContextTestSupport;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.component.mock.MockEndpoint;
+import org.apache.camel.impl.ThrottlingExceptionRoutePolicy;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ThrottlingExceptionRoutePolicyKeepOpenOnInitTest extends ContextTestSupport {
+
+    private String url = "seda:foo?concurrentConsumers=20";
+    private MockEndpoint result;
+    private int size = 5;
+
+    private ThrottlingExceptionRoutePolicy policy;
+
+    @Override
+    @Before
+    public void setUp() throws Exception {
+        this.createPolicy();
+
+        super.setUp();
+        this.setUseRouteBuilder(true);
+        result = getMockEndpoint("mock:result");
+        context.getShutdownStrategy().setTimeout(1);
+    }
+
+    protected void createPolicy() {
+        int threshold = 2;
+        long failureWindow = 30;
+        long halfOpenAfter = 1000;
+        boolean keepOpen = true;
+        policy = new ThrottlingExceptionRoutePolicy(threshold, failureWindow, halfOpenAfter, null, keepOpen);
+    }
+
+    @Test
+    public void testThrottlingRoutePolicyStartWithAlwaysOpenOn() throws Exception {
+
+        log.debug("---- sending some messages");
+        for (int i = 0; i < size; i++) {
+            template.sendBody(url, "Message " + i);
+            Thread.sleep(3);
+        }
+
+        // gives time for policy half open check to run every second
+        // and should not close b/c keepOpen is true
+        Thread.sleep(2000);
```

Comment:
Is there a way to not sleep for 2 seconds, eg this test takes 5+ seconds to run, and it would be nice with fast unit tests of camel-core.